python/dgl/core.py

```diff
@@ -231,7 +231,10 @@ def data_dict_to_list(graph, data_dict, func, target):
             for srctype, _, dsttype in graph.canonical_etypes:
                 if target == "u":
                     src_id = graph.get_ntype_id(srctype)
-                    output_list[src_id] = data_dict[srctype]
+                    if isinstance(data_dict, dict):
+                        output_list[src_id] = data_dict[srctype]
+                    else:
```

Comment:
what's the type of `data_dict` in `else`? tensor?

Reply:
> what's the type of `data_dict` in `else`? tensor?

yes, it's a tensor.

Reply:
> what about other branches like `target == 'v'` or any other code in `data_dict_to_list` which may hit the same issue here?

For the most outer branch(the else branch for `if isinstance(func, fn.BinaryMessageFunction):`, it already take a similar approach as mine.